### ISSUE TITLE
[BUG] Backlink search do not work well with Block IDs

### ISSUE THREAD

**Issue by rwaltr** [renerocksai/telekasten.nvim#103]
**Please confirm**
- [x] I am running the latest version of this plugin.

**Describe the bug**
Backlinks do not work well with blockID notation. Because the search for "Backlinks" contains `\[\[search\]\]`. This does not allow for notation like `[[file#^withblock]] notation to become a result.

**To Reproduce**
1. Start nvim.
2. create the following files.

- hello.md
```markdown
Hello world!
^a-block-id
```

- backlink.md
```markdown
I would like to link [[hello#^a-block-id]]
```

3. In `backlink.md` use `<cmd> lua require('telekasten').follow_link()<CR>`. This should send you to `hello.md` at `^a-block-id`.
4. Attempt to use `<cmd> lua require('telekasten').show_backlinks()<CR>`, your results should be zero.
5. you can remove `\]\]` to see the backlink.

**Expected behavior**
Step 4 should of resulted with at least `backlink.md`

**Operating system (please complete the following information):**
 - OS: Linux

**Additional context**
Config as of writing this report is located here https://github.com/rwaltr/dotfiles/blob/2b058a3076c9897ded7238c1d7797ea388682782/home/dot_config/nvim/lua/rwaltr/telekasten.lua

**Comment by wakatara** [comment]
This problem also happens with backlinks from links to headings in a file 

```
[[Journal 2022#2022-03-25 Fri]]
```

As well. I assume the fix is similar, we need to change the regex to allow for information before the `\]\]` section of the backlink search.

(and btw, just wanted to say how much I'm kinda digging telekasten. Lurving it enough to think about porting over from org-mode emacs. 😍)

**Comment by lambtho12** [comment]
I believe that the issue comes from the way telekasten determines if an object is a link or a tag. I discussed it a bit more in #110.
I think I am going to do a full rework of that function instead of adding bugfix after bugfix for each specific feature.

**Comment by lambtho12** [comment]
Closed in #111
